### PR TITLE
[Components with more than 1 service] LPS-185842 Refactor components from Commerce modules

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://issues.liferay.com/browse/LPS-185842

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:startup

File: modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/internal/target/ApplyToSKUCommerceDiscountTargetImpl.java [hunk 1]
```diff
@@ -45,7 +45,7 @@
 		"commerce.discount.target.key=" + CommerceDiscountConstants.TARGET_SKUS,
 		"commerce.discount.target.order:Integer=20"
 	},
-	service = {CommerceDiscountSKUTarget.class, CommerceDiscountTarget.class}
+	service = CommerceDiscountTarget.class
```
Comment (rafaprax) [inline modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/internal/target/ApplyToSKUCommerceDiscountTargetImpl.java:48]: Hey @MrJohn1911, the interface `CommerceDiscountSKUTarget` seems useless. Please remove that. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:commerce

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:commerce

**Comment by zhangyan0701** [comment]
Semantic versioning issue:
```
     [exec] > Task :apps:commerce:commerce-discount-api:baseline
     [exec]   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec] = ================================================== ========== ========== ========== ========== ==========
     [exec] * com.liferay.commerce.discount.target               MAJOR      1.1.0      1.1.0      2.0.0      VERSION INCREASE REQUIRED
     [exec] 	-   interface  com.liferay.commerce.discount.target.CommerceDiscountSKUTarget
     [exec] 		-   access     abstract
     [exec] 		-   method     contributeDocument(com.liferay.portal.kernel.search.Document,com.liferay.commerce.discount.model.CommerceDiscount)
     [exec] 			-   access     abstract
     [exec] 			-   return     void
     [exec] 		-   method     postProcessContextBooleanFilter(com.liferay.portal.kernel.search.filter.BooleanFilter,com.liferay.commerce.product.model.CPInstance)
     [exec] 			-   access     abstract
     [exec] 			-   return     void
     [exec] [Baseline Warning] Bundle Version Change Recommended: 17.0.0
     [exec] [Baseline Report] Mode: diff (persisted)
     [exec] Semantic versioning is incorrect while checking /opt/dev/projects/github/liferay-portal/tools/sdk/dist/com.liferay.commerce.discount.api-16.2.1.jar against /opt/dev/projects/github/liferay-portal/.gradle/caches/modules-2/files-2.1/com.liferay.commerce/com.liferay.commerce.discount.api/16.2.0/2db128d8278db6ef343efe86b9915adb8f7433ad/com.liferay.commerce.discount.api-16.2.0.jar
```
Full console log:
https://testray.liferay.com/reports/production/logs/2023-05/test-1-35/test-portal-acceptance-pullrequest(master)/4180/modules-semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0

Other failures are unrelated.

**Review by rafaprax — APPROVED** [review]

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
@MrJohn1911 No regressions found.

**Comment by MrJohn1911** [comment]
In the last push, I just did a baseline.

**Comment by dantewang** [comment]
@MrJohn1911 Could you extract common logic from implementations of `CommerceDiscountTarget` and maybe `CommerceDiscountProductTarget` to corresponding base classes to reduce repeating code? Thanks!

**Pull request closed without merging**